src/EnergyPlus/WeatherManager.cc

```diff
@@ -9990,15 +10002,17 @@ namespace WeatherManager {
                 env.ActualWeather = true;
             } else { // std RunPeriod
                 env.RollDayTypeOnRepeat = runPer.RollDayTypeOnRepeat;
-
                 if (env.StartYear == env.EndYear) {
                     // Short-circuit all the calculations, we're in a single year
-                    if (!WFAllowsLeapYears) {
-                        env.IsLeapYear = false; // explicit set
-                        LocalLeapYearAdd = 0;
-                    } else {
-                        env.IsLeapYear = true; // explicit set
-                        LocalLeapYearAdd = 1;
+                    LocalLeapYearAdd = 0;
+                    if (isLeapYear(env.StartYear)) {
+                        // If a leap year is supported by the weather file, do it.
+                        if (WFAllowsLeapYears) {
+                            env.IsLeapYear = true; // explicit set, this might be unwise
+                            LocalLeapYearAdd = 1;
+                        } else {
+                            env.IsLeapYear = false; // explicit set, this might be unwise
+                        }
```

Comment:
@jasondegraw hey, should we have put a warning in here for if the runperiod is a leap year but the weather file doesn't support it?  If so is this the right spot for it?  We'll add it within a different branch if so.  (Or you could add it to the namespace change branch on the next merge up)

Reply:
@Myoldmopar I'm not sure it is the right spot to add a warning. It would be good to have a warning, but my preference would be to wait until weather handling gets refactored. If we put in an issue on the lack of warning it'll be easier on whoever does the refactor. If we add the warning now, it'll be one more thing that someone will have to find. Previous behavior was to be silent, so I think it'd be alright to leave it alone for a bit longer, but if you'd rather add the warning that's OK too.

Reply:
Given how late in the game we are right now, I support the idea of doing nothing but opening up a ticket.  Thanks @jasondegraw